src/lib/reasoners/bitv_rel.ml

```diff
@@ -171,45 +173,67 @@ end = struct
     let changed = if !changed then SX.add r t.changed else t.changed in
     { changed; bitlists }
 
-  let get r t =
+  let update_signed ex { Bitv.value; negated } t bl =
+    let bl = if negated then Bitlist.lognot bl else bl in
+    update_leaf ex value t bl
+
+  let update ex r t bl =
+    fst @@ List.fold_left (fun (t, bl) { Bitv.bv; sz } ->
+        (* Extract the bitlist associated with the current component *)
+        let mid = Bitlist.width bl - sz in
+        let bl_tail =
+          if mid = 0 then Bitlist.empty else
+            Bitlist.extract bl 0 (mid - 1)
+        in
+        let bl = Bitlist.extract bl mid (Bitlist.width bl - 1) in
```

Comment:
I don't understand why we extract the end of the bistlist here instead of a prefix of it. I would expect to use a prefix of `bl` as domain to intersect with the current domain of `bv`.

Reply:
Bit-vectors are little endian, so if you have for instance `#b11110000` you might think of `#b1111` as being a prefix but the `#b1111` is actually the slice `4..7` (and `#b0000` is the slice `0..3`).

OTOH we represent compositions in the same order as for `concat` and so `(concat #b1111 #b0000) = #b11110000` is represented as the list `[#b1111; #b0000]` as a semantic value (well except that the `#b1111` and `#b0000` are actually merged during normalization but you get the idea).